### ISSUE TITLE
Warning in React 16.13.0

### ISSUE THREAD

**Issue by mikebarkmin** [zeit/swr#290]
Since React 16.13.0 is get this error:

```
Warning: Cannot update a component from inside the function body of a different component.
    in Unknown (created by Pagination)
    in div (created by ForwardRef(Grid))
    ...
```

It occurs when is use the `useSWRPages` hook. Everything seems to be working. But maybe this will lead to unwanted behavior.

Do you know what is causing this warning?

This is the component where is use `useSWRPages`:

```
function Pagination({
  name,
  pageSize,
  filter,
  order,
  orderBy,
  url,
  selector,
  Renderer,
  variant,
  spacing,
  placeholder
}) {
  const {
    pages,
    isEmpty,
    isLoadingMore,
    isReachingEnd,
    loadMore
  } = useSWRPages(
    name,
    ({ offset, withSWR }) => {
      const page = offset / pageSize;
      // eslint-disable-next-line react-hooks/rules-of-hooks
      const { data } = withSWR(
        useRequest(
          `${url}?${constructPagingQuery(
            page,
            pageSize,
            order,
            orderBy,
            humps.decamelizeKeys(filter)
          )}`
        )
      );

      if (!data) {
        return null;
      }

      if (variant == 'grid') {
        return selector(data).map(d => (
          <Grid item key={d.id} xs={12} sm={6} lg={3} xl={2}>
            <Renderer style={{ height: '100%' }} {...d} />
          </Grid>
        ));
      } else if (variant == 'list') {
        return selector(data).map(d => (
          <Grid item key={d.id} xs={12}>
            <Renderer style={{ height: '100%' }} {...d} />
          </Grid>
        ));
      }
    },
    (SWR, index) => {
      if (SWR.data && selector(SWR.data).length === 0) return null;
      return (index + 1) * pageSize;
    },
    [filter, order, orderBy, url]
  );

  return (
    <Box display="flex" flexDirection="column">
      {!isEmpty ? (
        <>
          <Grid
            container
            spacing={spacing}
            justify="center"
            alignItems="stretch"
          >
            {pages}
          </Grid>
          <Box
            marginTop={4}
            display="flex"
            flexDirection="column"
            alignItems="center"
          >
            <ProgressButton
              onClick={loadMore}
              disabled={isReachingEnd || isLoadingMore}
              loading={isLoadingMore}
            >
              {isLoadingMore
                ? '. . .'
                : isReachingEnd
                ? 'no more data'
                : 'load more'}
            </ProgressButton>
          </Box>
        </>
      ) : (
        placeholder
      )}
    </Box>
  );
}

```

**Comment by sergiodxa** [comment]
I think is because your props variant and selector, those are causing the individual pages to update and is causing the warning, but not sure how to solve that 🤔

**Comment by sergiodxa** [comment]
If you are not changing them then probably you need to wrap selector in React.useCallback to avoid re-creating the function on each render.

**Comment by baptistelemoine** [comment]
https://reactjs.org/blog/2020/02/26/react-v16.13.0.html#new-warnings

**Comment by YinghaoLiu-1112** [comment]
Have a same warning with use useSWRPages

**Comment by Ashot-KR** [comment]
Have the same.
Looks like it comes from `withSWR`

**Comment by Aryan-mor** [comment]
Have a same warning with use useSWRPages :(

**Comment by vishnugopal** [comment]
Same for me! React stack trace isolates it down to line 126 here:
![Screenshot 2020-03-21 at 10 57 49 AM](https://user-images.githubusercontent.com/533/77220108-dbfd5080-6b62-11ea-8ef2-e3bda2156c41.png)

**Comment by neolefty** [comment]
Yes, I also think it is from the new warnings — from calling `setPageSWRs()` inside `useSWRPages()`. That would mean it requires internal code changes, for example replacing with with `useEffect(() => setPageSWRs())` or `setTimeout(() => setPageSWRs())`.

**Comment by neolefty** [comment]
PR: https://github.com/zeit/swr/pull/317

**Comment by quietshu** [comment]
This issue should be fixed in `swr@0.2.0` 🎉 Huge thanks to @neolefty!
